### PR TITLE
(MAINT) Standardize docker test

### PR DESCRIPTION
**Pull request by nwolfe**
Use docker-compose so we can use our shared helper methods, and apply
other consistency updates.

----

Depends on https://github.com/puppetlabs/pupperware/pull/114

### CODE REVIEW

File: docker/Gemfile [hunk 1]
```diff
@@ -4,5 +4,5 @@ gem 'rspec'
 gem 'rspec_junit_formatter'
 gem 'pupperware',
     :git => 'https://github.com/puppetlabs/pupperware.git',
-    :branch => 'master',
+    :branch => 'fix-stderr-support',
```
Comment (underscorgan) [inline docker/Gemfile:7]: This should go back to master before merging

**Review by underscorgan — COMMENTED** [review]
We should revisit this if we ever end up with more complicated entrypoints, but this seems reasonable

**Comment by nwolfe** [comment]
Going to spend a few minutes seeing if we can't tease apart the entrypoint and command so we don't have to overwrite entrypoint in the tests.

**Comment by nwolfe** [comment]
Nevermind, not putting r10k on compose.

**Pull request closed without merging**